src/main/java/gui/TimetableBlock.java

```diff
@@ -156,7 +156,11 @@
             feedbackPane.setVisible(false);
             dragging = false;
             snapPane(thisBlock, dummyPane, e.getSceneX(), e.getSceneY(), draggingType);
-            this.fireEvent(new ShotblockUpdatedEvent());
+
+            if (parentBlock instanceof CameraShotBlock) {
+                this.fireEvent(new CameraShotBlockUpdatedEvent(parentBlock,
```

Comment:
Wouldn't it be better to extract shotblock event class to the parentBlock? (i.e. that ShotBlock has a method that returns a subclass of ShotBlockUpdatedEvent). This would cut back on instance of logic in a gui element.